### ISSUE TITLE
Feature Request: EC2 Autoscale 

### ISSUE THREAD

**Issue by adityacs** [influxdata/kapacitor#1603]
Creating this ticket to just start the discussion.

@nathanielc For implementing this can we use AWS "go sdk" or we have to do our own REST API implementation as in K8 and Docker?

**Comment by nathanielc** [comment]
I believe we already have the "go sdk" as a dependency for some EC2 scrapping logic. If you can use that dependency to implement EC2 autoscalling that would be perfect.

**Comment by adityacs** [comment]
@nathanielc Sure. Thanks

**Comment by adityacs** [comment]
@nathanielc What I realized is that, as in Kubernetes and Docker where we get "serviceName" tag to do "group by $serviceName" on the metrics. We don't have any tags like "autoscaling_group_name" in "System" metrics in telegraf. Without knowing, to which autoscaling_group the EC2 instances belong to, we don't have proper metrics in place to decide the auto scaling logic. 

I have opened a ticket in telegraf to discuss on this feature.[https://github.com/influxdata/telegraf/issues/3314](url)

**Comment by adityacs** [comment]
@nathanielc Need some help.

 I am trying to run "dep ensure" and I am getting below errors. Kindly help me in solving these errors.

```
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.11.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.11.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.10.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.10.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.9.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.9.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.8.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.8.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.7.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.7.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.6.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.6.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.5.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.5.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.4.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.4.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.3.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.3.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.2.1
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.2.1 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.2.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.2.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.1.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@0.1.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@v2.0.0-alpha.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@v2.0.0-alpha.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@v1.3.0-beta.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@v1.3.0-beta.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@v1.0.0-rc.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@v1.0.0-rc.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@master
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@master wants missing rev bd73d950fa4440dae889bd9917bff7cef539f86e of github.com/Azure/azure-sdk-for-go
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@cancel-scrape-contexts
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@cancel-scrape-contexts not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@dev-2.0
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@dev-2.0 not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@fossa
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@fossa not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@go-targetmanager-wait
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@go-targetmanager-wait not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@improve-consoles
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@improve-consoles not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@log-query-stats
(40)  ✗   github.com/prometheus/prometheus (from github.com/goller/prometheus)@log-query-stats not allowed by constraint logger-targetmanager-wait:
(40)      logger-targetmanager-wait from (root)
(39)      try github.com/prometheus/prometheus (from github.com/goller/prometheus)@logger-targetmanager-wait
(39)  ✓ select github.com/prometheus/prometheus (from github.com/goller/prometheus)@logger-targetmanager-wait w/25 pkgs
(40)  ? revisit github.com/aws/aws-sdk-go to add 4 pkgs
(40)    ✓ include 24 more pkgs from github.com/aws/aws-sdk-go@v1.8.16
(40)  ? revisit golang.org/x/sys to add 1 pkgs
(41)    ✓ include 1 more pkgs from golang.org/x/sys@master
(40)  ? revisit github.com/prometheus/common to add 2 pkgs
(42)    ✓ include 4 more pkgs from github.com/prometheus/common@master
(40)  ? attempt github.com/Azure/azure-sdk-for-go with 2 pkgs; at least 1 versions to try
(43)      try github.com/Azure/azure-sdk-for-go@bd73d950fa4440dae889bd9917bff7cef539f86e
(43)  ✗   unable to update checked out version: : command failed: [git checkout bd73d950fa4440dae889bd9917bff7cef539f86e]: exit status 1
(43)      try github.com/Azure/azure-sdk-for-go@v1.2
(44)  ✗   github.com/Azure/azure-sdk-for-go@v1.2 not allowed by constraint bd73d950fa4440dae889bd9917bff7cef539f86e:
(44)      bd73d950fa4440dae889bd9917bff7cef539f86e from github.com/prometheus/prometheus (from github.com/goller/prometheus)@logger-targetmanager-wait

```

**Comment by adityacs** [comment]
Closed in #1622
